repositories/app_repository.go

```diff
@@ -0,0 +1,99 @@
+package repositories
+
+import (
+	"context"
+	"errors"
+
+	workloadsv1alpha1 "code.cloudfoundry.org/cf-k8s-controllers/api/v1alpha1"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type AppRepo struct{}
+
+const (
+	StartedState DesiredState = "STARTED"
+	StoppedState DesiredState = "STOPPED"
+)
+
+type AppRecord struct {
+	Name      string
+	GUID      string
+	SpaceGUID string
+	State     DesiredState
+	Lifecycle Lifecycle
+	CreatedAt string
+	UpdatedAt string
+}
+
+type DesiredState string
+
+type Lifecycle struct {
+	Data LifecycleData
+}
+
+type LifecycleData struct {
+	Buildpacks []string
+	Stack      string
+}
+
+// TODO: Make a general ConfigureClient function / config and client generating package
+func (f *AppRepo) ConfigureClient(config *rest.Config) (client.Client, error) {
+	client, err := client.New(config, client.Options{Scheme: scheme.Scheme})
+	if err != nil {
+		return nil, err
+	}
+	return client, nil
+}
+
+func (f *AppRepo) FetchApp(client client.Client, appGUID string) (AppRecord, error) {
+	// TODO: Could look up namespace from guid => namespace cache to do Get
+	appList := &workloadsv1alpha1.CFAppList{}
+	err := client.List(context.Background(), appList)
```

Comment:
I think this will always error for a user without a ClusterRoleBinding allowing list on CFApp.

At the moment, only users with admin, admin-read-only or global-auditor roles will have this permission. All org and space related roles will only have namespace level permissions to list on CFApp.

It might be worth using a user with only namespaced permissions in the integration tests to catch this.

Reply:
@kieron-dev we're trying to think about how this could all work in the "final" implementation:

I'm imagining the following:

1. The shim has a privileged k8s client of its own that it can use for various things (populate lookup caches, etc.) and we could use this client to list all namespaces
2. We instantiate the k8s client that handler uses on each request based on the `Authentication` header's token. This is what is actually used to fetch the resources we'll display. We do this per namespace that we looked up with the privileged client.

It's hard to do this right now since we haven't implemented the second part of that yet. Do you know which story that `Authentication` header stuff falls under? Is it https://github.com/cloudfoundry/cf-k8s-api/issues/10?

Reply:
There's a proposal for modifying the CF CLI to send the k8s auth credentials. Not sure there are stories for the work yet.